packages/office-ui-fabric-react/src/utilities/selection/SelectionZone.tsx

```diff
@@ -489,6 +488,27 @@ export class SelectionZone extends BaseComponent<ISelectionZoneProps, {}> {
     this._clearAndSelectIndex(index);
   }
 
+  /**
+   * To avoid high startup cost of traversing the DOM on component mount,
+   * defer finding the scrollable parent until a click interaction.
+   *
+   * The styles will probably already calculated since we're running in a click handler,
+   * so this is less likely to cause layout thrashing then doing it in mount.
+   *
+   * @param ev the click event
+   */
+  private _findScrollParentAndTryClearOnEmptyClick(ev: MouseEvent) {
+    const scrollParent = findScrollableParent(this._root.current);
+    // unbind this handler and replace binding with a binding on the actual scrollable parent
+    this._events.off(document, 'click', this._findScrollParentAndTryClearOnEmptyClick);
+    this._events.on(scrollParent, 'click', this._tryClearOnEmptyClick);
+
+    // If we clicked inside the scrollable parent, call through to the handler on this click.
+    if ((scrollParent && ev.target instanceof Node && scrollParent.contains(ev.target)) || scrollParent === ev.target) {
```

Comment:
Is the complex if statement needed? I am not sure it was done in the previous implementation and I don't understand the consequences.

Reply:
The instanceof check is needed so we can check contains (though maybe findscrollable parent's contract could be tightened).

Because the first click handler is on the doc root, we have to check the event target contains / is equal to the scroll parent on first click to know if we want to act as if we had assigned the click handler to that element on Mount.

Reply:
(tl;dr it's to ensure that the first click also clears the selection _if_ it was in the scrollparent we just found)